### ISSUE TITLE
Zookeeper not starting if only `ZOOKEEPER_SECURE_CLIENT_PORT` but not `ZOOKEEPER_CLIENT_PORT` set

### ISSUE THREAD

**Issue by haeuserd** [confluentinc/kafka-images#92]
I want to use Zookeeper with only the TLS port enabled and the non-TLS port disabled. However, if I only specify `ZOOKEEPER_SECURE_CLIENT_PORT` but not `ZOOKEEPER_CLIENT_PORT`, Zookeeper startup fails with the following error:
```
===> User
uid=1000(appuser) gid=1000(appuser) groups=1000(appuser)
===> Configuring ...
ZOOKEEPER_CLIENT_PORT is required.
Command [/usr/local/bin/dub ensure ZOOKEEPER_CLIENT_PORT] FAILED !
```

I would suggest to fix this by ensuring at least one of the two variables is set:
```diff
diff --git a/zookeeper/include/etc/confluent/docker/configure b/zookeeper/include/etc/confluent/docker/configure
index 742207d9..2c2af7d1 100755
--- a/zookeeper/include/etc/confluent/docker/configure
+++ b/zookeeper/include/etc/confluent/docker/configure
@@ -16,7 +16,7 @@
 
 . /etc/confluent/docker/bash-config
 
-dub ensure ZOOKEEPER_CLIENT_PORT
+dub ensure-atleast-one ZOOKEEPER_CLIENT_PORT ZOOKEEPER_SECURE_CLIENT_PORT
 
 dub path /etc/kafka/ writable
 

```

In order to work, this also requires changes in the cp-base-new base image: confluent-docker-utils needs to be updated to v0.0.45 because the currently used version 0.0.43 is suffering from [a bug in `dub ensure-atleast-one`](
https://github.com/confluentinc/confluent-docker-utils/pull/20).

**Comment by jheinitz** [comment]
Hello!

Did you find a workaround for this in the meantime? We are facing exactly the same problem.

Kind regards

Jens

**Comment by haeuserd** [comment]
Hi Jens,
my workaround was to build my own zookeeper image from `confluentinc/cp-zookeeper` in which I replaced the `/zookeeper/include/etc/confluent/docker/configure` file with a version that omits the check.

**Comment by jheinitz** [comment]
Hi!

In the meantime I figured out by myself that setting `ZOOKEEPER_CLIENT_PORT=0` disables the plaintext port, but allows the docker container to start successfully. Maybe this is also a valid workaround for you.

Kind regards

Jens

**Comment by haeuserd** [comment]
Fixed by #140
